src/Views/Display.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+namespace srag\Plugins\OpencastPageComponent\Views;
+
+use ILIAS\UI\Factory;
+use ILIAS\UI\Component\Component;
+use srag\Plugins\Opencast\Container\Container;
+use srag\Plugins\Opencast\Model\Event\EventAPIRepository;
+use srag\Plugins\Opencast\Model\Event\Event;
+use srag\Plugins\Opencast\Model\Config\PluginConfig;
+use srag\Plugins\OpencastPageComponent\Authorization\TokenRepository;
+use srag\Plugins\Opencast\Model\Publication\Config\PublicationUsage;
+
+/**
+ * @author Fabian Schmid <fabian@sr.solutions>
+ */
+class Display implements ViewElement
+{
+    private const PLUGIN_DIRECTORY = './Customizing/global/plugins/Services/COPage/PageComponent/OpencastPageComponent';
+    private Factory $ui_factory;
+    private EventAPIRepository $event_repository;
+    private ?Event $event = null;
+    private float $publication_ratio = 16 / 9;
+    private ?string $error = null;
+
+    public function __construct(
+        private array $properties,
+        private Container $container,
+        private string $mode
+    ) {
+        $this->ui_factory = $this->container->ilias()->ui()->factory();
+        $event_repository = $this->container->get(EventAPIRepository::class);
+        $this->container->ilias()->ui()->mainTemplate()->addCss(
+            self::PLUGIN_DIRECTORY . '/templates/css/presentation.css'
+        );
+        try {
+            $this->event = $event_repository->find(
+                $this->properties[\ilOpencastPageComponentPluginGUI::PROP_EVENT_ID] ?? ''
+            );
+            $event_ratio = (new EventDimensions($event_repository))->determineForEvent(
+                $this->event
+            );
+
+            $this->publication_ratio = $event_ratio?->getRatio() ?? $this->publication_ratio;
+        } catch (\Throwable $e) {
+            $this->error = $e->getMessage();
+        }
+    }
+
+    public function get(): Component|array
+    {
+        $as_link = (bool) ($this->properties[\ilOpencastPageComponentPluginGUI::PROP_AS_LINK] ?? false);
+        $max_width = (string) ($this->properties[\ilOpencastPageComponentPluginGUI::PROP_WIDTH] ?? '');
+        $max_width = $max_width !== '' ? $max_width . 'px' : 'auto';
+
+        $configured_aspect_ratio = (float) ($this->properties[\ilOpencastPageComponentPluginGUI::PROP_ASPECT_RATIO] ?? Edit::RATIO_AS_PUBLICATION);
```

Comment:
`Edit::RATIO_AS_PUBLICATION` as fallback and then casting to `float` would make the `===` comparison in next line alway `false`, therefore the (float) casting must be limited to the `$this->properties[\ilOpencastPageComponentPluginGUI::PROP_ASPECT_RATIO]`

```suggestion
        $configured_aspect_ratio =  ((float) $this->properties[\ilOpencastPageComponentPluginGUI::PROP_ASPECT_RATIO]) ?? Edit::RATIO_AS_PUBLICATION);
```